beacon-chain/blockchain/error.go

```diff
@@ -41,18 +41,25 @@ var (
 // The block violates certain fork choice rules (before finalized slot, not finalized ancestor)
 type invalidBlock struct {
 	error
+	root [32]byte
 }
 
 type invalidBlockError interface {
 	Error() string
 	InvalidBlock() bool
+	BlockRoot() [32]byte
 }
 
 // InvalidBlock returns true for `invalidBlock`.
 func (e invalidBlock) InvalidBlock() bool {
 	return true
 }
 
+// BlockRoot returns the invalid block root.
+func (e invalidBlock) BlockRoot() [32]byte {
```

Comment:
Why is this needed instead of just returning `true` in the last line of `IsInvalidBlock`? (like 72)